scylla-cdc-lib/src/main/java/com/scylladb/cdc/lib/CDCConsumer.java

```diff
@@ -188,15 +177,27 @@ public Builder withSleepAfterExceptionMs(long sleepAfterExceptionMs) {
             return this;
         }
 
-        private static int getDefaultWorkersCount() {
-            int result = Runtime.getRuntime().availableProcessors() - 1;
-            return result > 0 ? result : 1;
+        public Builder withExecutorService(ScheduledExecutorService executorService) {
+            this.executorService = executorService;
+            return this;
         }
 
         public CDCConsumer build() {
+            if (executorService == null) {
+                final ScheduledThreadPoolExecutor s = new ScheduledThreadPoolExecutor(
+                        workersCount);
+                s.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
+                executorService = s;
+            }
+            workerConfigurationBuilder.withExecutorService(executorService);
             return new CDCConsumer(cqlConfigurationBuilder.build(),
                     masterConfigurationBuilder, workerConfigurationBuilder,
-                    consumerProvider, workersCount);
+                    consumerProvider);
+        }
+
+        private static int getDefaultWorkersCount() {
```

Comment:
Why moving the method if it wasn't changed?

Reply:
It wasn't moved. Code was inserted before it and git decided to make a weird diff.

Reply:
It seems that it was moved. It was before `public CDCConsumer build() {` and now it's after so one of those methods had to be moved, no?

Reply:
Hm, probably because it is a private helper, and the others are public attribute/build methods.

Reply:
I.e. good form. :-)